### PR TITLE
User Story 60501: Azure API Management

### PR DESCRIPTION
**Pull request by TomArcherMsft**
Part of POC to test generating sample code and articles using OpenAI.

### CODE REVIEW

File: quickstart/101-azure-api-management-create/providers.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">=0.12"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>2.0"
```
Comment (lonegunmanb) [inline quickstart/101-azure-api-management-create/providers.tf]: Should we use `"~>3.0"` instead?

Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/providers.tf:6]: Yes. This is just the first checkin of an auto-generated sample. I haven't made any edits yet.

Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/providers.tf]: Fixed

**Review by TomArcherMsft — COMMENTED** [review]
Thanks for the review! All issues addressed.

**Comment by TomArcherMsft** [comment]
@lonegunmanb This PR is ready for review/merge. Thanks.

File: quickstart/101-azure-api-management-create/main.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
```
Comment (grayzu) [inline quickstart/101-azure-api-management-create/main.tf:5]: No need to change it in these examples but you should not be using the type for the name of the resources. You should use something that represents that specific resource or example or this.

File: quickstart/101-azure-api-management-create/main.tf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "random_string" "name" {
```
Comment (grayzu) [inline quickstart/101-azure-api-management-create/main.tf]: Should update the name to specify what this string is used for. "api_service_name" or something like that.

File: quickstart/101-azure-api-management-create/providers.tf [hunk 4]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">=0.12"
```
Comment (grayzu) [inline quickstart/101-azure-api-management-create/providers.tf]: Why are we only requiring 0.12 for this example. That is super old. Should update to version 1.0 at least.

File: quickstart/101-azure-api-management-create/providers.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>3.0"
```
Comment (grayzu) [inline quickstart/101-azure-api-management-create/providers.tf:6]: Will this example really work with any version for the provider 3.0? If not, it should be updated to a more current version.

File: quickstart/101-azure-api-management-create/variables.tf [hunk 6]
```diff
@@ -0,0 +1,49 @@
+variable "resource_group_location" {
+  default     = "eastus"
```
Comment (grayzu) [inline quickstart/101-azure-api-management-create/variables.tf:3]: Add types to variables.

**Review by grayzu — CHANGES_REQUESTED** [review]
Thanks! Looking good. Few comments.

File: quickstart/101-azure-api-management-create/main.tf (hunk 3 above)
Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/main.tf]: Fixed

File: quickstart/101-azure-api-management-create/providers.tf (hunk 4 above)
Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/providers.tf]: Fixed

File: quickstart/101-azure-api-management-create/providers.tf (hunk 5 above)
Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/providers.tf:6]: Fixed

File: quickstart/101-azure-api-management-create/variables.tf (hunk 6 above)
Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/variables.tf:3]: Fixed

**Review by lonegunmanb — APPROVED** [review]
One more issue to solve

File: quickstart/101-azure-api-management-create/providers.tf [hunk 7]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">=1.00"
```
Comment (lonegunmanb) [inline quickstart/101-azure-api-management-create/providers.tf]: Could we use the following expression?:

```hcl
required_version = ">=1.0"
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Hi @TomArcherMsft, just find one more issue to solve

File: quickstart/101-azure-api-management-create/providers.tf (hunk 7 above)
Comment (TomArcherMsft) [reply quickstart/101-azure-api-management-create/providers.tf]: Fixed.

**Comment by TomArcherMsft** [comment]
@lonegunmanb I've made the edits. Thanks!

**Pull request merged**